Sources/AsyncKit/ConnectionPool/EventLoopConnectionPool.swift

```diff
@@ -197,9 +204,19 @@ public final class EventLoopConnectionPool<Source> where Source: ConnectionPoolS
             // connections are exhausted, we must wait for one to be returned
             logger.debug("Connection pool exhausted on this event loop, adding request to waitlist")
             let promise = eventLoop.makePromise(of: Source.Connection.self)
-            self.waiters.append((logger, promise))
+            let uuid = UUID()
+            self.waiters.append((logger, promise, uuid))
+            
+            let task = eventLoop.scheduleTask(in: self.creationTimeout) { [weak self, logger, promise] in
```

Comment:
Why do `logger` and `promise` need to be captured by the block?

Reply:
`logger` is needed for the `.error` logging. `promise` is used to fail the promise if the timeout is over & (after latest push) for the `===` comparison.

Reply:
Yes, but why can't those be captured implicitly?

Reply:
Good question. Not sure why I did this in this way 🤔. I can remove the explicit capture if preferred. CC @tanner0101